sklearn/utils/tests/test_pprint.py

```diff
@@ -574,3 +574,36 @@ def set_params(self, **params):
     with config_context(print_changed_only=False):
         expected = "WithKWargs(a='something', b='unchanged', c='abcd', d=None)"
         assert expected == est.__repr__()
+
+def test_complexity_print_changed_only():
+    # Make sure `__repr__` is called the same amount of times
+    # whether `print_changed_only` is True or False
+    # Non-regression test for
+    # https://github.com/scikit-learn/scikit-learn/issues/18490
+
+    class ReprCounter(BaseEstimator):
+        nb_times_repr_called = 0
+        def __repr__(self):
+            ReprCounter.nb_times_repr_called += 1
+            return super().__repr__()
+
+    class DummyEstimator(TransformerMixin, ReprCounter):
+        def __init__(self, estimator=None):
+            self.estimator = estimator
+
+        def transform(self, X, copy=None):
+            return self
```

Comment:
We don't need this because we never call `fit` on the pipeline

Reply:
When `transform` was not there, I had this error : 
```
E               TypeError: All intermediate steps should be transformers and implement fit and transform or be the string 'passthrough' 'DummyEstimator(estimator=DummyEstimator())' (type <class 'sklearn.utils.tests.test_pprint.test_complexity_print_changed_only.<locals>.DummyEstimator'>) doesn't
```

Reply:
I am also going to add `estimator.transform(None)` because a test coverage test has failed due to this function not being tested.

Reply:
It's preferable to just add  `# pragma: no cover` instead